my-app/src/test/java/com/JavaWebServer/app/OptionsParserTest.java

```diff
@@ -2,15 +2,21 @@
 import static org.hamcrest.CoreMatchers.instanceOf;
 import static org.junit.Assert.assertEquals;
 import org.junit.Test;
-import java.util.*;
+import java.util.HashMap;
+import java.util.Map;
+
 public class OptionsParserTest  {
-  @Test 
+ @Test 
  public void parseArugmentsTest() {
    String [] keys = {"-p","-d"};
    String [] testArguements = {"-p","9090","-d","PUBLIC_DIR"};
-   HashMap <String,String> expectedResults = new HashMap<String, String>();
+   Map <String,String> expectedResults = new HashMap<String, String>();
+   Map <String, String> parsedArguments = OptionsParser.parse(testArguements, keys);
+
    expectedResults.put("-p", "9090");
-   expectedResults.put("-d","PUBLIC_DIR"); 
-   assertEquals(OptionsParser.parse(testArguements, keys), expectedResults);
+   expectedResults.put("-d","PUBLIC_DIR");
```

Comment:
I would recommend grouping your setup together so that you can clearly see what is expected or pre-configured, then what action is being tested, then your assertions. 

```java
String [] supportedKeys = {"-p", "-d"};
String [] inputArguments = {"-p","9090","-d","PUBLIC_DIR"};

Map<String, String> expectedResults = new HashMap<>();
expectedResults.put("-p", "9090");
expectedResults.put("-d", "PUBLIC_DIR");

Map<String, String> parsedArguments = OptionsParser.parse(inputArguments, supportedKeys);

assertEquals(parsedArguments.keySet(), expectedResults.keySet());
```